external_search.py

```diff
@@ -1702,6 +1719,13 @@ class Filter(SearchBase):
     # the useful information from the search engine isn't lost.
     KNOWN_SCRIPT_FIELDS = ['last_update']
 
+    # In general, someone looking for things "by this person" is
+    # probably looking for one of these roles.
+    AUTHOR_MATCH_ROLES = list(Contributor.AUTHOR_ROLES) + [
+        Contributor.EDITOR_ROLE, Contributor.DIRECTOR_ROLE,
+        Contributor.ACTOR_ROLE
+    ]
+
```

Comment:
I don't think we want to change AUTHOR_ROLES here. The editor of a book is not the "author", but if you're looking for "books by this person" then books that they edited should quality.

Reply:
Aren't the contributor feeds only used for authors anyway? This seems like it would be more useful for the regular keyword search. Narrator might also be a good one.

Reply:
I think we're just a couple branches away from being able to use the contributors subdocument in searches. Like, we could add a hypothesis now that gives a big boost if an author-role contributor is an exact match for the search term, with an even bigger boost for primary author.

Then we could change the name fields so they're indexed twice, once as a keyword and once as a text field (this part seems really annoying). At that point we don't need to search on sort_author anymore; we just need it for searching.